pallets/living-assets-evolution/src/lib.rs

```diff
@@ -34,36 +34,80 @@ pub mod pallet {
 	pub trait Config: frame_system::Config {
 		/// Because this pallet emits events, it depends on the runtime's definition of an event.
 		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+		/// Limit for the length of `token_uri`
+		#[pallet::constant]
+		type MaxTokenUriLength: Get<u32>;
 		/// Type representing the weight of this pallet
 		type WeightInfo: WeightInfo;
+		/// Converts [`AccountId`] to [`H160`]
+		type AccountIdToH160: Convert<AccountIdOf<Self>, H160>;
 	}
 
 	/// Collection counter
 	#[pallet::storage]
 	#[pallet::getter(fn collection_counter)]
 	pub(super) type CollectionCounter<T: Config> = StorageValue<_, CollectionId, ValueQuery>;
 
-	// storage for the ownership of collections
+	/// Storage for the ownership of collections
 	#[pallet::storage]
 	#[pallet::getter(fn collection_owner)]
 	pub type CollectionOwner<T: Config> =
-		StorageMap<_, Blake2_128Concat, CollectionId, T::AccountId, OptionQuery>;
+		StorageMap<_, Blake2_128Concat, CollectionId, AccountIdOf<T>, OptionQuery>;
 
-	// Pallets use events to inform users when important changes are made.
-	// https://docs.substrate.io/main-docs/build/events-errors/
+	/// Minted assets
+	#[pallet::storage]
```

Comment:
What about having just a storage item like:
```rust
pub type Asset<T: Config> = StorageDoubleMap<
		_,
		Blake2_128Concat,
		CollectionId,
		Blake2_128Concat,
		TokenId,
		TokenInfo<T>,
		OptionQuery,
	>;
```
where `TokenInfo` is a struct that includes `uri` and `owner` as fields?

Reply:
responded in Discord